src/csm_testing/tests/sat_functional/hwinv/test_hwinv.py

```diff
@@ -0,0 +1,389 @@
+        command = 'sat hwinv --list-node-bmcs'
+        self.run_hwinv_test(command, 'node bmcs')
+
+    def test_hwinv_list_router_bmcs(self) -> None:
+        """Test that `sat hwinv --list-router-bmcs` returns proper header"""
+        command = 'sat hwinv --list-router-bmcs'
+        self.run_hwinv_test(command, 'router bmcs')
+
+    def test_hwinv_list_mgmt_switches(self) -> None:
+        """Test that `sat hwinv --list-mgmt-switches` returns proper header"""
+        command = 'sat hwinv --list-mgmt-switches'
+        self.run_hwinv_test(command, 'mgmt switches')
+
+    def test_hwinv_list_cabinet_pdus(self) -> None:
+        """Test that `sat hwinv --list-cabinet-pdus` returns proper header"""
+        command = 'sat hwinv --list-cabinet-pdus'
+        self.run_hwinv_test(command, 'cabinet pdus')
+
+    def test_hwinv_list_cabinet_pdu_power_connectors(self) -> None:
+        """Test that `sat hwinv --list-cabinet-pdu-power-connectors` returns proper header"""
+        command = 'sat hwinv --list-cabinet-pdu-power-connectors'
+        self.run_hwinv_test(command, 'cabinet pdu power connectors')
+
+    def test_hwinv_list_all(self) -> None:
+        """Test that `sat hwinv --list-all --show-missing --show-empty` validates only the components present in the system."""
+        # Get the list of components present in the system
+        present_components = get_present_components()
+
+        command = 'sat hwinv --list-all --show-missing --show-empty'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stdout_lines = stdout.splitlines()
+
+        # Iterate through the lines to find sections and validate headers
+        for i, line in enumerate(stdout_lines):
+            # Match section titles like "Listing of all <SOME COMPONENT TYPE>"
+            match = re.match(r"Listing of all (.+)", line.strip())
+            if not match:
+                continue
+
+            section_name = match.group(1)
+            lower_sec   = section_name.lower()
+            # Skip any section type not reported as present
+            if lower_sec not in present_components:
+                continue
+
+            self.assertIn(section_name, HWINV_HEADERS,
+                f"Unexpected section '{section_name}' found in output."
+            )
+
+            # Extract the expected header
+            expected_header = HWINV_HEADERS[section_name]
+
+            # Get the header line (3 lines below the section title)
+            header_line_index = i + 3
+            if header_line_index >= len(stdout_lines):
+                self.fail(f"Header row for section '{section_name}' not found in output.")
+            header_line = stdout_lines[header_line_index].strip()
+
+            # Check for empty tables (no header or data rows)
+            if not header_line.startswith("|"):
+                continue
```

Comment:
If there are any components of the given type in the output of `cray hsm inventory hardware list`, then we should expect there to be a non-empty table here, and we should fail if the table is missing.